rflx/graph.py

```diff
@@ -1,216 +1,148 @@
-
-        result = Dot(graph_name=f'"{name}"')
-        result.set_graph_defaults(
-            splines="true", ranksep="0.1 equally", pad="0.1", truecolor="true", bgcolor="#00000000"
-        )
-        result.set_edge_defaults(
-            fontname="Fira Code", fontcolor="#6f6f6f", color="#6f6f6f", penwidth="2.5"
-        )
-        result.set_node_defaults(
-            fontname="Arimo",
-            fontcolor="#ffffff",
-            color="#6f6f6f",
-            fillcolor="#009641",
-            width="1.5",
-            style='"rounded,filled"',
-            shape="box",
+    msg = copy(message)
+    if not msg.structure:
+        # https://github.com/Componolit/RecordFlux/issues/643
+        # pylint: disable-next = protected-access
+        msg._structure = [Link(INITIAL, FINAL)]
+
+    result = _graph_with_defaults(msg.full_name)
+    result.add_node(
+        Node(name="Initial", fillcolor="#ffffff", shape="circle", width="0.5", label="")
+    )
+    for f in msg.fields:
+        result.add_node(Node(name=f.name))
+    for i, l in enumerate(msg.structure):
+        intermediate_node = f"intermediate_{i}"
+        result.add_node(
+            Node(
+                name=intermediate_node,
+                label=_edge_label(l),
+                style="",
+                fontname="Fira Code",
+                fontcolor="#6f6f6f",
+                color="#6f6f6f",
+                penwidth="0",
+                width="0",
+                height="0",
+            )
         )
-        return result
+        result.add_edge(Edge(src=l.source.name, dst=intermediate_node, arrowhead="none"))
+        result.add_edge(Edge(src=intermediate_node, dst=l.target.name, minlen="1"))
+    result.add_node(Node(name="Final", fillcolor="#6f6f6f", shape="circle", width="0.5", label=""))
+    return result
+
+
+def create_session_graph(session: AbstractSession, ignore: Sequence[str] = None) -> Dot:
+    """
+    Return pydot graph representation of session.
 
-    def _add_state(self, state: State, result: Dot, variables: Counter[ID]) -> None:
+    If present, the ignore parameter contains a list of regular expression which are matched
+    against state names and the names of edge targets in order. If a regular expression matches
+    a state name or an edge target name, the respective state or edge is excluded from the result
+    graph.
+    """
 
-        assert isinstance(self._data, AbstractSession)
+    def _is_ignored(name: ID) -> bool:
+        if not ignore:
+            return False
+        for regex in ignore:
+            if re.search(regex, str(name), re.IGNORECASE):
+                return True
+        return False
 
-        height = sqrt(self._degree[state.identifier.name] + 1)
-        width = 1.3 * sqrt(self._degree[state.identifier.name] + 1)
-        variables_read: Counter[ID] = collections.Counter()
-        variables_write: Counter[ID] = collections.Counter()
+    result = _graph_with_defaults("Session")
```

Comment:
Shouldn't the session identifier be used here instead of `"Session"`?

Reply:
Good catch! Changed.